addons/crm/tests/test_crm_lead.py

```diff
@@ -496,3 +496,23 @@ def test_phone_mobile_search(self):
         self.assertEqual(lead_1, self.env['crm.lead'].search([
             ('phone_mobile_search', 'like', '+32485001122')
         ]))
+
+    def test_no_date_closed_update_between_won_stages(self):
+        # Test for one won lead
+        stage_team1_won2 = self.env['crm.stage'].create({
+            'name': 'Won2',
+            'sequence': 75,
+            'team_id': self.sales_team_1.id,
+            'is_won': True,
+        })
+        won_lead = self.lead_team_1_won
+        date_closed = won_lead.date_closed
+        won_lead.stage_id = stage_team1_won2
+        self.assertEqual(date_closed, won_lead.date_closed)
+
+        # Test for one won and one unwon lead
+        leads = won_lead + self.lead_1
+        self.assertFalse(self.lead_1.date_closed)
+        leads.stage_id = self.stage_team1_won
+        self.assertEqual(date_closed, won_lead.date_closed)
+        self.assertTrue(self.lead_1.date_closed)
```

Comment:
I fear this test is either non deterministic, either not really testing the feature due to timing during test. You will have to use freezetime (from freeze_gun) to clearly test the update of date_closed. You should have some existing examples in the code. You can do `` with freezetime(my_reference): # do stuff`` to cleanly mock datetimes.

Note that SQL now does not use datetime library and has to be patched independently (notably for create_date) but I don't think you will have to do it in 14, as date_closed uses fields.Datetime.now() which is based on datetime library.